### PR TITLE
chore: added missing packages for initializedTooLateHeuristic and optimised tests

### PR DESCRIPTION
**Pull request by kirrg001**

### CODE REVIEW

**Comment by basti1302** [comment]
> the regex never matched

There is an integration test at https://github.com/instana/nodejs/tree/main/packages/collector/test/tracing/misc/too_late that has been green so far. What exactly do you mean with "the regex never matched"?

**Comment by kirrg001** [comment]
nvm the function is working fine :D Added rdkafka and a unit test. I'd suggest to keep a single case in the integration test (it runs 26s) and rely on the unit test for the rest.

**Comment by basti1302** [comment]
> nvm the function is working fine :D Added rdkafka and a unit test. I'd suggest to keep a single case in the integration test (it runs 26s) and rely on the unit test for the rest.
> 

Yes, great idea!

**Comment by kirrg001** [comment]
> Yes, great idea!

Yeah 👍

File: packages/core/test/util/initializedTooLateHeuristic_test.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    });
+
+    require.cache['/@contrast/agent/'] = {};
+
+    const tracing = require('../../src/');
+    tracing.init();
+  });
+
+  after(() => {
+    delete require.cache['/@contrast/agent/'];
```
Comment (basti1302) [inline packages/core/test/util/initializedTooLateHeuristic_test.js:31]: _question:_ Do we also need to undo the `sinon.stub` calls on the `requireHook` module in `after` here`? I think we should do that.

File: packages/core/test/util/initializedTooLateHeuristic_test.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+  it('hasBeenInitializedTooLate is true', () => {
+    const originalRequireCache = require.cache;
+
+    instrumentedModules.forEach(moduleName => {
+      // eslint-disable-next-line no-console
+      console.log(`for module ${moduleName}`);
+
+      // RESET CACHE & module
+      require.cache = originalRequireCache;
```
Comment (basti1302) [inline packages/core/test/util/initializedTooLateHeuristic_test.js:57]: _question:_ Does that actually do anything? Aren't `require.cache` and `originalRequireCache` two references to the very same object? AFAIK `require.resolve` does not create a new object for `require.cache`, it just updates the existing data structure right? (Also, our own modifications will not replace require.cache with a new object, as we only set entries and delete them.)

**Review by basti1302 — APPROVED** [review]
LGTM!

File: packages/core/test/util/initializedTooLateHeuristic_test.js (hunk 1 above)
Comment (kirrg001) [reply packages/core/test/util/initializedTooLateHeuristic_test.js:31]: TODO: check if `sinon.reset` is needed

File: packages/core/test/util/initializedTooLateHeuristic_test.js (hunk 2 above)
Comment (kirrg001) [reply packages/core/test/util/initializedTooLateHeuristic_test.js:57]: Check if we need to use clone

**Pull request merged**